scripts/download-rc.js

```diff
@@ -0,0 +1,136 @@
+'use strict';
+const fs = require('fs');
+const os = require('os');
+const {spawnSync} = require('child_process');
+const rcParams = require('./rc-params');
+
+const ON_WINDOWS = os.platform() === 'win32';
+const HAZELCAST_ENTERPRISE_KEY = process.env.HAZELCAST_ENTERPRISE_KEY ? process.env.HAZELCAST_ENTERPRISE_KEY : '';
+
+let REPO;
+let ENTERPRISE_REPO;
+let TEST_REPO;
+let ENTERPRISE_TEST_REPO;
+
+if (rcParams.HZ_VERSION.endsWith('-SNAPSHOT')) {
+    REPO = rcParams.SNAPSHOT_REPO;
+    ENTERPRISE_REPO = rcParams.ENTERPRISE_SNAPSHOT_REPO;
+} else {
+    REPO = rcParams.RELEASE_REPO;
+    ENTERPRISE_REPO = rcParams.ENTERPRISE_RELEASE_REPO;
+}
+
+if (rcParams.HZ_TEST_VERSION.endsWith('-SNAPSHOT')) {
+    TEST_REPO = rcParams.SNAPSHOT_REPO;
+    ENTERPRISE_TEST_REPO = rcParams.ENTERPRISE_SNAPSHOT_REPO;
+} else {
+    TEST_REPO = rcParams.RELEASE_REPO;
+    ENTERPRISE_TEST_REPO = rcParams.ENTERPRISE_RELEASE_REPO;
+}
+
+if (fs.existsSync(`hazelcast-remote-controller-${rcParams.HAZELCAST_RC_VERSION}.jar`)) {
```

Comment:
nit: I'd rather wrap everything starting from this line and below into a function, so that consumer module can call it explicitly (instead of implicit initialization on `require`). As a side effect, this way will make it easier to handle exceptions if necessary.